src/core/pal/feature.cpp

```diff
@@ -1231,99 +1209,121 @@ int FeaturePart::createCurvedCandidatesAlongLine( QList< LabelPosition* >& lPos,
     flags = FLAG_ON_LINE; // default flag
   // placements may need to be reversed if using line position dependent orientation
   // and the line has right-to-left direction
-  bool reversed = ( !( flags & FLAG_MAP_ORIENTATION ) ? isRightToLeft : false );
+  bool reversed = (( flags & FLAG_MAP_ORIENTATION ) ? isRightToLeft : false );
```

Comment:
I think it's this change which is causing the incorrectly flipped labels. I was also very confused by this flag (it seems backwards to me too!),but it must be getting misinterpreted when being used later.